### PR TITLE
Issue2

### PR DESCRIPTION
**Pull request by Alekearl**

### CODE REVIEW

File: src/main/java/core/basesyntax/Circle.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Circle extends Figure implements Area, ColorCreator {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Circle.java:5]: make your Figure implement Area and you won't need to do it in each class

File: src/main/java/core/basesyntax/Circle.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        this.radius = radius;
+    }
+
+    @Override
+    public double countArea() {
+        return Math.PI * (radius * radius);
+    }
+
+    @Override
+    public Color getColor() {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Circle.java:26]: this method should return value of color field and not generte random color each

File: src/main/java/core/basesyntax/ColorCreator.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface ColorCreator {
+    Color getColor();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/ColorCreator.java:4]: make it as a simple getter inside of abstract class

File: src/main/java/core/basesyntax/Figure.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    public abstract String onScreen();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:4]: this class should have common state (fields), for example you can store color here

Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:4]: change method name, it's not clear what it does and it should be a verb. Plus, move it to a separate interface

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Circle.java (hunk 1 above)
Comment (Alekearl) [reply src/main/java/core/basesyntax/Circle.java:5]: Done. =)

File: src/main/java/core/basesyntax/Circle.java (hunk 2 above)
Comment (Alekearl) [reply src/main/java/core/basesyntax/Circle.java:26]: Changed

File: src/main/java/core/basesyntax/ColorCreator.java (hunk 3 above)
Comment (Alekearl) [reply src/main/java/core/basesyntax/ColorCreator.java:4]: Changed

File: src/main/java/core/basesyntax/Figure.java (hunk 4 above)
Comment (Alekearl) [reply src/main/java/core/basesyntax/Figure.java:4]: Changed

Comment (Alekearl) [reply src/main/java/core/basesyntax/Figure.java:4]: Changed

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**